src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+     */
+    constexpr BaseKey(const char* str_) : _str{str_} {}
+
+    /**
+     * @brief return property key string
+     * @return Pointer to const string key representation
+     */
+    const char* str() const {
+        return _str;
+    }
+
+    /**
+     * @brief compares property key string
+     * @return true if string is the same
+     */
+    bool operator==(const std::string& str) const {
+        return _str == str;
+    }
+
+    /**
+     * @brief compares property key string
+     * @return true if string is the same
+     */
+    friend bool operator==(const std::string& str, const BaseKey<T, mutability_>& key) {
+        return key == str;
+    }
+
+private:
+    const char* _str = nullptr;
+};
+template <typename T, PropertyMutability M>
+inline std::ostream& operator<<(std::ostream& os, const BaseKey<T, M>& key) {
+    return os << key.str();
+}
+}  // namespace util
+/** @endcond */
+
+/**
+ * @brief This class us used to bind property key string with paramter type
+ * @tparam T type of parameter used to pass or get property
+ */
+template <typename T, PropertyMutability mutability_ = PropertyMutability::RW>
+struct Key : public util::BaseKey<T, mutability_> {
+    using util::BaseKey<T, mutability_>::BaseKey;
+    /**
+     * @brief Constructs property
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of string key representation and type erased paramter.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {this->str(), Any::make<T>(std::forward<Args>(args)...)};
+    }
+};
+
+/**
+ * @brief This class us used to bind property key string with read only paramter type
+ * @tparam T type of parameter used to pass or get property
+ */
+template <typename T>
+struct Key<T, PropertyMutability::RO> : public util::BaseKey<T, PropertyMutability::RO> {
+    using util::BaseKey<T, PropertyMutability::RO>::BaseKey;
+};
+
+/**
+ * @brief Read only property to get a std::vector<PropertyName> of supported read only properies.
+ *
+ * This can be used as an compiled model property as well.
+ *
+ */
+static constexpr Key<std::vector<PropertyName>, PropertyMutability::RO> supported_properties{"SUPPORTED_PROPERTIES"};
+
+/**
+ * @brief Read only property to get a std::vector<std::string> of available device IDs
+ */
+static constexpr Key<std::vector<std::string>, PropertyMutability::RO> available_devices{"AVAILABLE_DEVICES"};
+
+/**
+ * @brief Read only property to get a std::vector<std::string> of optimization options per device.
+ */
+static constexpr Key<std::vector<std::string>, PropertyMutability::RO> optimization_capabilities{
+    "OPTIMIZATION_CAPABILITIES"};
+
+/**
+ * @brief Read only property which defines support of import/export functionality by plugin
+ */
+static constexpr Key<bool, PropertyMutability::RO> import_export_support{"IMPORT_EXPORT_SUPPORT"};
+
```

Comment:
please, also add:
- RANGE_FOR_STREAMS
- RANGE_FOR_ASYNC_INFER_REQUESTS
- OPTIMAL_BATCH_SIZE
- CPU_THREADS_NUM

Reply:
@myshevts Please have a look seems that RANGE_FOR_STREAMS RANGE_FOR_ASYNC_INFER_REQUESTS can be deprecated and replaced with `ov::hint::num_requests` note it is `RW` property for plugin and may be `ro` for `compiled_model`
CPU_THREADS_NUM - should be replaced with ov::execution::concurancy. Note if we are going to use tbb executor in next release we can not control number of threads

Reply:
added